product-cli/cmd/nodepool/create.go

```diff
@@ -0,0 +1,45 @@
+package nodepool
+
+import (
+	"github.com/spf13/cobra"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1beta1"
+	"github.com/openshift/hypershift/cmd/nodepool/core"
+	hyperShiftKubeVirt "github.com/openshift/hypershift/cmd/nodepool/kubevirt"
+	"github.com/openshift/hypershift/product-cli/cmd/nodepool/kubevirt"
+)
+
+// The following lines are needed in order to validate that any platform implementing PlatformOptions satisfy the interface
+var _ core.PlatformOptions = &hyperShiftKubeVirt.KubevirtPlatformCreateOptions{}
+
+func NewCreateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "nodepool",
+		Short:        "Create a Hosted Cluster NodePool",
+		SilenceUsage: true,
+	}
+
+	opts := &core.CreateNodePoolOptions{
+		Name:            "example",
+		Namespace:       "clusters",
+		ClusterName:     "example",
+		NodeCount:       2,
+		ReleaseImage:    "",
+		NodeUpgradeType: hyperv1.UpgradeTypeReplace,
+		Arch:            "amd64",
+	}
+
+	cmd.PersistentFlags().StringVar(&opts.Name, "name", opts.Name, "The name of the NodePool")
+	cmd.PersistentFlags().StringVar(&opts.Namespace, "namespace", opts.Namespace, "The namespace in which to create the NodePool")
+	cmd.PersistentFlags().Int32Var(&opts.NodeCount, "node-count", opts.NodeCount, "The number of nodes to create in the NodePool")
+	cmd.PersistentFlags().StringVar(&opts.ClusterName, "cluster-name", opts.ClusterName, "The name of the HostedCluster nodes in this pool will join")
+	cmd.PersistentFlags().StringVar(&opts.ReleaseImage, "release-image", opts.ReleaseImage, "The release image for nodes. If empty, defaults to the same release image as the HostedCluster.")
+	cmd.PersistentFlags().Var(&opts.NodeUpgradeType, "node-upgrade-type", "The NodePool upgrade strategy for how nodes should behave when upgraded. Supported options: Replace, InPlace")
+	cmd.PersistentFlags().StringVar(&opts.Arch, "arch", opts.Arch, "The processor architecture for the NodePool (e.g. arm64, amd64)")
```

Comment:
I'm planning to leave it for now unless you really want it gone. It will be needed for the AWS NodePool. We also have validation that would block this being used on KubeVirt [here](https://github.com/openshift/hypershift/blob/3cb3166f62ccc03b6fb346cff904b6427a7dcb5c/hypershift-operator/controllers/nodepool/nodepool_controller.go#L446).

Reply:
oh yeah, this was for the generic nodepool case and not just KubeVirt. makes sense.